src/lib/path.js

```diff
@@ -104,3 +104,9 @@ export function join (a, b) {
   if (b.startsWith('/')) b = b.slice(1)
   return `${a}/${b}`
 }
+
+export function dirname (path) {
```

Comment:
Let's just use node core `path` stuff for this kind of thing, https://nodejs.org/api/path.html#path_path_dirname_path

Reply:
Of note this file was intended to be a lib for ipfs/ipld path helpers, rather than any sort of path.